pkg/client/apiutil/restmapper_test.go

```diff
@@ -210,6 +210,28 @@ func TestLazyRestMapperProvider(t *testing.T) {
 		g.Expect(crt.GetRequestCount()).To(gmg.Equal(4))
 	})
 
+	t.Run("LazyRESTMapper KindsFor should work correctly with empty versions list", func(t *testing.T) {
+		g := gmg.NewWithT(t)
+
+		httpClient, err := rest.HTTPClientFor(restCfg)
+		g.Expect(err).NotTo(gmg.HaveOccurred())
+
+		crt := newCountingRoundTripper(httpClient.Transport)
+		httpClient.Transport = crt
+
+		lazyRestMapper, err := apiutil.NewDynamicRESTMapper(restCfg, httpClient)
+		g.Expect(err).NotTo(gmg.HaveOccurred())
+
+		g.Expect(crt.GetRequestCount()).To(gmg.Equal(0))
+
+		kinds, err := lazyRestMapper.KindsFor(schema.GroupVersionResource{Group: "autoscaling", Resource: "horizontalpodautoscaler"})
+		g.Expect(err).NotTo(gmg.HaveOccurred())
+		g.Expect(len(kinds)).To(gmg.Equal(2))
```

Comment:
This test still fails because the result here is `4`. I do in general not think that its safe to assume that this restmapper will do the same thing as the upstream one if you don't call it with all of group, version and kind.

Reply:
Thanks, removed assertion of result list size 😊